eva-ingest-remapped-variants/src/main/java/uk/ac/ebi/eva/ingest/remapped/configuration/batch/io/VcfReaderConfiguration.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2021 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.ingest.remapped.configuration.batch.io;
+
+import org.springframework.batch.core.configuration.annotation.StepScope;
+import org.springframework.batch.item.ItemStreamReader;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+
+import uk.ac.ebi.eva.commons.batch.io.AggregatedVcfLineMapper;
+import uk.ac.ebi.eva.commons.batch.io.UnwindingItemStreamReader;
+import uk.ac.ebi.eva.commons.batch.io.VcfReader;
+import uk.ac.ebi.eva.commons.core.models.Aggregation;
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+import uk.ac.ebi.eva.ingest.remapped.configuration.InputParametersConfiguration;
+import uk.ac.ebi.eva.ingest.remapped.parameters.InputParameters;
+
+import java.io.File;
+import java.io.IOException;
+
+import static uk.ac.ebi.eva.ingest.remapped.configuration.BeanNames.VCF_READER;
+
+@Configuration
+@Import(InputParametersConfiguration.class)
+public class VcfReaderConfiguration {
+
+    @Bean(VCF_READER)
+    @StepScope
+    public ItemStreamReader<Variant> unwindingReader(VcfReader vcfReader) {
+        return new UnwindingItemStreamReader<>(vcfReader);
+    }
+
+    @Bean
+    public VcfReader vcfReader(InputParameters inputParameters) throws IOException {
+        File vcfFile = new File(inputParameters.getVcf());
+        AggregatedVcfLineMapper lineMapper = new AggregatedVcfLineMapper("dummyFile", "dummyStudy", Aggregation.BASIC,
+                                                                         null);
```

Comment:
Are these dummy entries supposed to be there ?

Reply:
The file and study field are requested in the line mapper because this is also used when populating the `variant warehouse`. Those fields goes into the fields `files.fid` and `files.sid` in the variant warehouse but since that is not relevant for this pipeline we are just passing a dummy string as they are required.